app/models/azure_engine.rb

```diff
@@ -0,0 +1,154 @@
+require 'azure'
+require 'azure-contrib'
+require 'json'
+
+class AzureEngine
+  class Config
+    include ActiveSupport::Configurable
+    config_accessor :bucket_name
+    config_accessor :image_bucket_name
+    config_accessor :cdn_base_url
+    config_accessor :queue_name
+    config_accessor :azure_storage_access_key
+    config_accessor :azure_storage_account_name
+  end
+
+  def self.configure(&block)
+    yield config
+    Azure.config.storage_account_name = @config.azure_storage_account_name
+    Azure.config.storage_access_key = @config.azure_storage_access_key
+    @blob_client = Azure.blobs
+  end
+
+  def self.config
+    @config ||= Config.new
+  end
+
+  def self.queue_endpoint
+    url = "https://#{@config.azure_storage_account_name}.queue.core.windows.net/"
+    url
+  end
+
+  def self.resource_endpoint
+    unless @config.cdn_base_url.blank?
+      url = @config.cdn_base_url
+    else
+      url = "https://#{@config.azure_storage_account_name}.blob.core.windows.net/#{@config.image_bucket_name}"
+    end
+    url
+  end
+
+  def self.upload_endpoint
+    url = "https://#{@config.azure_storage_account_name}.blob.core.windows.net/#{@config.bucket_name}"
+    url
+  end
+
+  def self.send_message(message)
+    queues = Azure.queues
+    resp = queues.create_message(@config.queue_name, message)
+    resp
+  end
+
+  def self.receive_message(max_number = 10)
+    queues = Azure.queues
+    res = queues.list_messages(@config.queue_name, 600, {:number_of_messages => max_number})
+    res
+  end
+
+  def self.message_exist?(resp)
+    if !resp || resp.count == 0
+      false
+    else
+      true
+    end
+  end
+
+  def self.delete_message(message_object)
+    queues = Azure.queues
+    resp = queues.delete_message(@config.queue_name, message_object.id, message_object.pop_receipt)
+    resp
+  end
+
+  def self.batch_delete(entries)
+    entries.each do |entry|
+      self.delete_message(entry)
+    end
+  end
+
+  ## Blob
+  def self.upload_files(container, files, prefix)
+    files.each do |f|
+      if File.exist?(f)
+        content = File.open(f, 'rb') { |file| file.read }
```

Comment:
__[rubocop]__
* Style/SymbolProc: Pass `&:read` as an argument to `open` instead of a block.